gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/db/FabricDatabase.java

```diff
@@ -1773,7 +1773,7 @@ public LanguageConnectionContext setupConnection(ContextManager cm,
     // check if the user schema is a proper one else create a proper schema for
     // the user on the fly and add to DataDictionary
     if (defaultSchema.getUUID() == null) {
-      setupDefaultSchema(this.dd, lctx, tc, lctx.getAuthorizationId(), false);
+      setupDefaultSchema(this.dd, lctx, tc, lctx.getAuthorizationId().replace('-', '_'), false);
```

Comment:
See snappydata PR comment about possibility of clashes? Can't we continue to use '-' here and map that to something like five dashes only on hive side (and back if required)? This might be more work, so perhaps just document that clashes can happen if names contain both dashes and underscores for now.